store/src/commonMain/kotlin/org/mobilenativefoundation/store/store5/impl/FetcherController.kt

```diff
@@ -55,7 +57,8 @@ internal class FetcherController<Key : Any, Network : Any, Output : Any, Local :
      */
     private val sourceOfTruth: SourceOfTruthWithBarrier<Key, Network, Output, Local>?,
 
-    private val converter: Converter<Network, Output, Local>? = null
+    private val converter: Converter<Network, Output, Local>? = null,
+    private val processor: Processor<Output>? = null
```

Comment:
rather than changing FetcherController, how do you feel about making ProcessingFetcherController that can delegate to Fetcher controller and wrap with the additional functionality?
The advantage would be a switch in the store of which fetcher controller to use rather than having nullables in here